### DISCUSSION TITLE
Is there a way to find out which function block the tokio schedule

### DISCUSSION THREAD

**Discussion by attila-lin** [tokio-rs/console#410] [Q&A] [answered]
The users may misuse some block functions without `spawn_blocking` and they may cause the schedule to be blocked.

Currently, it is hard to debug.

**Comment by hds** [comment, marked as answer]
In tokio-console you can see from the poll time which **task** is blocking (because its busy time keeps going up), but it's not possible to get function level information.

You can use the (unstable) [task builder API](https://docs.rs/tokio/latest/tokio/task/struct.Builder.html) in tokio to name tasks and thus get a better idea of where the blocking is coming from.

**Comment by gftea** [comment]
@hds , is the busy, sched, idle are showing accumulative value but not a certain time window?

**Reply by hds** [reply]
The times shown are the sum over the entire lifetime of the task.

**Reply by gftea** [reply]
Thanks @hds , is there any way to reset the measure at live time?

**Reply by hds** [reply]
Not currently. No. If there is a specific action you want to measure separately, you could work around this by spawning that action as a new task.

If you have a specific use case in mind that you think would be useful for others too, feel free to open an Issue describing it.
